Fix the accents and some other things within gr-bb.ctb.

DaveMielke opened the pull request:
Fix the accents and some other things within gr-bb.ctb

bertfrees commented:
Hi Dave. Thanks for this.

I didn't want to bother you with it beforehand because I thought you knew the drill, but: some regression tests would be nice. Just a list of all the letters is probably enough, although some examples that show why we need the Latin letter definitions would be nice too maybe. 

I wonder why `capsletter` is `6` but the Greek uppercase letters are all defined as `letter` with the cap sign (46) integrated. Wouldn't it make more sense to make `capsletter 46`, like is done in el.ctb? And why are the Latin letters even needed?

Finally, I was wondering what explains the big differences between ancient and modern Greek braille, and why a different braille code is needed. Wikipedia says that there are two Greek braille codes: "Modern Greek Braille" used in Greece, and "International Greek Braille" for Greek letters or words used in mathematics or otherwise embedded in English and other languages. But it doesn't mention a braille code specially for ancient Greek. Do you know more about this?

DaveMielke commented:
[quoted lines by Bert Frees on 2019/07/20 at 04:48 -0700]

>I didn't want to bother you with it beforehand because I thought you knew the
>drill, but: some regression tests would be nice. Just a list of all the
>letters is probably enough, although some examples that show why we need the
>Latin letter definitions would be nice too maybe.

Yes, I'll get to it. I just wanted to get the changes in right away. I'm now
working on a few outstanding fixes to el.ctb. After they're done, I'll have a
look at adding test cases.

>I wonder why `capsletter` is `6` but the Greek uppercase letters are all
>defined as `letter` with the cap sign (46) integrated. Wouldn't it make more
>sense to make `capsletter 46`, like is done in el.ctb? And why are the Latin
>letters even needed?

I don't know and just left that part of the table as it was. Now that the Greek
part of it is fixed, I'll look into some of these other questions. My main
goal, at this point, was not wanting to break it for uses that I'm unaware of.

>Finally, I was wondering what explains the big differences between ancient and
>modern Greek braille, and why a different braille code is needed. 

el.ctb actually works much better for ancient Greek. The difference between the
two tables, in my opinion, has much more to do with the typical arrogance of
those whose native language is English. :-) The real difference between the two
tables is that el.ctb is for real Greek people and that gr-bb.ctb was developed
by english-speaking researchers who wanted to be able to read Greek but who
didn't bother to check with Greece first.

-- 
I believe the Bible to be the very Word of God: http://Mielke.cc/bible/
Dave Mielke            | 2213 Fox Crescent | WebHome: http://Mielke.cc/
EMail: Dave@Mielke.cc  | Ottawa, Ontario   | Twitter: @Dave_Mielke
Phone: +1 613 726 0014 | Canada  K2A 1H7   |

bertfrees commented:
>> I wonder why `capsletter` is `6` but the Greek uppercase letters are all defined as `letter` with the cap sign (46) integrated. Wouldn't it make more sense to make `capsletter 46`, like is done in el.ctb? And why are the Latin letters even needed?
>
> I don't know and just left that part of the table as it was. Now that the Greek part of it is fixed, I'll look into some of these other questions. My main goal, at this point, was not wanting to break it for uses that I'm unaware of.

OK. At some point I think a refactoring would be appropriate. It seems the starting point of this table was an English table which would explain why the Greek letter are "secondary". We shouldn't be too afraid to break things anyway, but with some added tests I think we're pretty safe.

>> Finally, I was wondering what explains the big differences between ancient and modern Greek braille, and why a different braille code is needed. 
>
> el.ctb actually works much better for ancient Greek. The difference between the two tables, in my opinion, has much more to do with the typical arrogance of those whose native language is English. :-) The real difference between the two tables is that el.ctb is for real Greek people and that gr-bb.ctb was developed by english-speaking researchers who wanted to be able to read Greek but who didn't bother to check with Greece first.

OK. So in that case, "international Greek" is actually a better description of the table than "ancient Greek" or "biblical Greek"?

DaveMielke commented:
[quoted lines by Bert Frees on 2019/07/20 at 07:07 -0700]

>At some point I think a refactoring would be appropriate. It seems the
>starting point of this table was an English table which would explain why the
>Greek letter are "secondary". 

I think that's exactly right. I just want to solve one problem at a time - the
most important of which was the extreme difficulty to actually read Greek with
it because the accents were messed up. :-) Another thing that needs to be done
is to ensure that back translating actually ends up typing Greek characters
(not English ones).

>We shouldn't be too afraid to break things anyway, but with some added tests I
>think we're pretty safe.

Yes. Don't worry - I'll get to the tests. My priority, right now, is not
wanting to prolong my need for Maria's time.

>OK. So in that case, "international Greek" is actually a better description of
>the table than "ancient Greek" or "biblical Greek"?

Yes, except that that might still encourage its use. I've already been trying
to think of a good name that means something like "englishified" without being
too rude. :-) I've considered el-us, as everyone is used to that country going
its own way on such things, but that wouldn't be accurate since UK people have
been involved as well.

As part of this, I'm wondering if you'd be willing to rename gr-bb to whatever
we decide is a better name. It really should be el-something as el is the
language code - GR is the territory code.

-- 
I believe the Bible to be the very Word of God: http://Mielke.cc/bible/
Dave Mielke            | 2213 Fox Crescent | WebHome: http://Mielke.cc/
EMail: Dave@Mielke.cc  | Ottawa, Ontario   | Twitter: @Dave_Mielke
Phone: +1 613 726 0014 | Canada  K2A 1H7   |

bertfrees commented:
Sure, we can rename the file if we find a more suitable name, but actually  don't really care about the file name. I would like the table metadata to be correct though. Currently the "display-name" field says "Ancient Greek braille" and "locale" says "grc". Possibly it can stay "grc" if the main or only application of this braille code is for historical texts. But the display name should be a bit more specific. I've also been thinking for a while about a new metadata field to specify on which language community a braille code applies (English in this case), which is not necessarily the same as the primary target language of the table (Greek in this case), which is what the "locale" field is for. In some cases this information can be encoded within the locale, more specifically in the region subtag. For example, the Liblouis table for transcribing Dutch texts and that is used in Belgium (Dutch is spoken in the Netherlands and Belgium) is tagged "nl-BE". However we can not use this trick for the "grc-en" ("el-en") case because that's not a valid language tag. And "grc-US" is a not accurate enough. A new field would solve the issue.

DaveMielke commented:
I've added a paragraph based on what Maria wrote that describes what the table
really is. I've also modified the comment on the first line. Let me know if
you'd like it to be more informative. No change to the display-name yet.

-- 
I believe the Bible to be the very Word of God: http://Mielke.cc/bible/
Dave Mielke            | 2213 Fox Crescent | WebHome: http://Mielke.cc/
EMail: Dave@Mielke.cc  | Ottawa, Ontario   | Twitter: @Dave_Mielke
Phone: +1 613 726 0014 | Canada  K2A 1H7   |

bertfrees commented:
Thanks. Maybe the name of the friend of Maria from Oxford should be mentioned? Also Maria's name and your name should be in the copyright lines.

DaveMielke commented:
Can the system handle Maria's name being written in Greek (as in her email
header), or should it be written using Latin letters?

-- 
I believe the Bible to be the very Word of God: http://Mielke.cc/bible/
Dave Mielke            | 2213 Fox Crescent | WebHome: http://Mielke.cc/
EMail: Dave@Mielke.cc  | Ottawa, Ontario   | Twitter: @Dave_Mielke
Phone: +1 613 726 0014 | Canada  K2A 1H7   |

bertfrees commented:
Yes you can use any characters. There are also Chinese names in some tables.

bertfrees commented on tables/gr-bb.ctb:
```diff
@@ -1,7 +1,7 @@
 #
 # -----------
-#-display-name: Ancient Greek braille
+#-display-name: Greek braille used by some scholars
```
What's important I think is the fact that the table used by non-Greek speaking. That it is for studying is less relevant. Its useful information but I would drop it from the "display-name" metadata because this is what is presented to users and those descriptions should be as concise and accurate as possible.

DaveMielke commented:
[quoted lines by Bert Frees on 2019/07/21 at 16:00 -0700]

>What's important I think is the fact that the table used by non-Greek
>speaking. 

to me, that's a tough one as I don't think we want to be encouraging non-Greek
people to be using it at all. One of it's "advantages" is that each accent is a
separate character and that there are no diphthong contractions. It's too bad
that some of the letters are represented differently, but we could say
something like:

   simplified international Greek 

And then, if we do that, what about maybe calling the table el-simlified.ctb?

-- 
I believe the Bible to be the very Word of God: http://Mielke.cc/bible/
Dave Mielke            | 2213 Fox Crescent | WebHome: http://Mielke.cc/
EMail: Dave@Mielke.cc  | Ottawa, Ontario   | Twitter: @Dave_Mielke
Phone: +1 613 726 0014 | Canada  K2A 1H7   |

bertfrees commented:
> but we could say something like "simplified international Greek".

Sounds good. But I wonder if we also need to indicate that it is for English speaking or whether this doesn't matter.

> what about maybe calling the table el-simplified.ctb?

OK.

Note that I've used the ".utb" extension because it's not contracted. Or you can use the new ".tbl" extension because that is how we are eventually gonna name all the tables anyway (when all the metadata is correct and we have found a good naming scheme).

DaveMielke commented:
[quoted lines by Bert Frees on 2019/07/22 at 02:30 -0700]

>I wonder if we also need to indicate that it is for English speaking or
>whether this doesn't matter.

For the display-name, should the language be first, as it's more important? If
so, What'd your convention be for adding the modifiers?

   Greek - simplified international
   Greek, simplified international
   Greek (siplified international)
   ...

>Note that I've used the ".utb" extension because it's not contracted. Or you
>can use the new ".tbl" extension because that is how we are eventually gonna
>name all the tables anyway (when all the metadata is correct and we have found
>a good naming scheme).

I think I'll go with .utb for now. I think I'll also split the Greek part out
into el-simplified.uti so that, if anyone should ever want to, it'll be easy to
include elsewhere.

If we want to put English in the name, would el-simplified-en.utb be okay?

-- 
I believe the Bible to be the very Word of God: http://Mielke.cc/bible/
Dave Mielke            | 2213 Fox Crescent | WebHome: http://Mielke.cc/
EMail: Dave@Mielke.cc  | Ottawa, Ontario   | Twitter: @Dave_Mielke
Phone: +1 613 726 0014 | Canada  K2A 1H7   |

bertfrees commented:
File name doesn't matter that much to me, but it should probably reflect what we put in the metadata. The metadata should either get a `#+system` field which is an identifier of the braille system, or/and if appropriate, the new field that I suggested to indicate the target audience.

By the way I would keep the `#+locale: grc`. If we limit it to grc we discourage the use of this table more.

There is not really a convention for adding modifiers at the moment. There is an overview of all the display names in [extra/generate-display-names/display-names](https://github.com/liblouis/liblouis/blob/master/extra/generate-display-names/display-names). You can see that there are a few occasions where something is added in brackets. The language is always first, except in the case of "Two-cell Chinese braille". Maybe this could be changed to "Chinese braille (two-cell)".

DaveMielke commented:
[quoted lines by Bert Frees on 2019/07/22 at 05:08 -0700]

>The metadata should either get a `#+system` field which is an identifier of
>the braille system,

Isn't that what #+name already is?

I wish this braille code had a name, but, as far as I know, it doesn't.

>or/and if appropriate, the new field that I suggested to indicate the target
>audience.

Are there other tables which could use such a property? What are some examples?

>By the way I would keep the `#+locale: grc`. 

Yes, since Greek is its purpose anyway. Why grc instead of el, though?

I wonder if there's an ISO country code for "any country", "international", or
some such thing.

>If we limit it to grc we discourage the use of this table more.

Do you think it's a bad idea, then, for me to split the Greek part out into a
.uti file?

-- 
I believe the Bible to be the very Word of God: http://Mielke.cc/bible/
Dave Mielke            | 2213 Fox Crescent | WebHome: http://Mielke.cc/
EMail: Dave@Mielke.cc  | Ottawa, Ontario   | Twitter: @Dave_Mielke
Phone: +1 613 726 0014 | Canada  K2A 1H7   |

bertfrees commented:
>> The metadata should either get a `#+system` field which is an identifier of the braille system,
>
> Isn't that what #+name already is?
>
> I wish this braille code had a name, but, as far as I know, it doesn't.

In this case, it's maybe not an official system, but we can still give it our own name, like e.g. `grc-international`. By the way, I just had a look at the "[World Braille Usage, third edition](http://liblouis.org/braille-specs/world-braille-usage-third-edition.pdf)" document and it has a chapter for "Greek (international)" on page 185.

`#+name` is kind of the same, yes, but it is a description in words rather than a computer readable identifier (and it can also contain more information then just the braille system; one system can have several variants and grades).

> Are there other tables which could use such a property? What are some examples?

Not at the moment, except the example I gave earlier with Dutch in Belgium vs. The Netherlands. We haven't really got different braille systems, but some small differences exist anyway (actually, handling of Greek is one of those differences), and although efforts are being done to create a common standard, we currently still have two Liblouis tables. As I said before this difference can be encoded within the `#+locale` field (`nl-BE` vs. `nl-NL`), but if a separate field would exist for the target audience, `#+locale` would become `nl` for both (and I would probably rename the field to `#+language`), and the new field would become `nl-BE` vs. `nl-NL`. I guess BANA vs. BAUK is another example.

>> By the way I would keep the `#+locale: grc`. 
> 
> Yes, since Greek is its purpose anyway. Why grc instead of el, though?

Exactly to limit the scope. `grc` means classical greek: https://iso639-3.sil.org/code/grc.

> I wonder if there's an ISO country code for "any country", "international", or some such thing.

Don't think there is such a thing.

> Do you think it's a bad idea, then, for me to split the Greek part out into a .uti file?

I still like the idea. At least it won't hurt. And it could be useful for removing duplication between tables. Exactly which tables could make use of it I don't know. Maybe we should first investigate what various tables have in common. I know many languages use a "alphabet switch indicator" which complicates things because Liblouis can't handle that very well. This will be improved when https://github.com/liblouis/liblouis/issues/702 is implemented.

DaveMielke commented:
[quoted lines by Bert Frees on 2019/07/22 at 08:30 -0700]

>In this case, it's maybe not an official system, but we can still give it our
>own name, like e.g. `grc-international`. 

That's what I just renamed the table to (grc-international.utb). Should I add a
#+name too?

The display-name is now "Greek international braille".

>By the way, I just had a look at the "[World Braille Usage, third
>edition](http://liblouis.org/braille-specs/world-braille-usage-third-edition.pdf)"
>document and it has a chapter for "Greek (international)" on page 185.

Yes, that's the right code, so I guess we now know its name. It raises a
question, however. It has two schemes for the accented letters. One prefixes a
letter with a dot combination representing the accent. The other uses a set of
single braille characters - one for each accent+letter combination. If we
define both then we'd need two names - not just one.

>> Do you think it's a bad idea, then, for me to split the Greek part out into a .uti file?
>
>I still like the idea. At least it won't hurt. And it could be useful for
>removing duplication between tables. Exactly which tables could make use of it
>I don't know. 

I figured that it'd just make it easy for whoever to include it on top of
his/her primary language table.

-- 
I believe the Bible to be the very Word of God: http://Mielke.cc/bible/
Dave Mielke            | 2213 Fox Crescent | WebHome: http://Mielke.cc/
EMail: Dave@Mielke.cc  | Ottawa, Ontario   | Twitter: @Dave_Mielke
Phone: +1 613 726 0014 | Canada  K2A 1H7   |

bertfrees commented:
> That's what I just renamed the table to (grc-international.utb). Should I add a #+name too?

Sure.

Regarding the different schemes for accents: hmm yeah, it seems it depends on the country. Since the table you're currently working on is the English version, probably only one accent scheme is required. But if you decide to split off the Greek letters into a "uti" version, maybe it makes sense two make two versions of it, yes.

DaveMielke commented:
[quoted lines by Bert Frees on 2019/07/22 at 10:51 -0700]

>Regarding the different schemes for accents: hmm yeah, it seems it depends on
>the country. 

Did I miss something in the document? Does it actually say that one scheme is
meant for some languages and the other scheme for other languages?

>But if you decide to split off the Greek letters into a "uti" version, maybe
>it makes sense two make two versions of it, yes.

But how to name them? Do you see anything in the document that gives each
scheme some kind of a different name?

-- 
I believe the Bible to be the very Word of God: http://Mielke.cc/bible/
Dave Mielke            | 2213 Fox Crescent | WebHome: http://Mielke.cc/
EMail: Dave@Mielke.cc  | Ottawa, Ontario   | Twitter: @Dave_Mielke
Phone: +1 613 726 0014 | Canada  K2A 1H7   |

bertfrees commented:
Yes, the document says:

> The single accented letters may be used in some countries, while others use 2-cell characters.

To come up with a concise name for these is a difficult one. Maybe "one-cell-accents" vs. "two-cell-accents". Or "accents-composed" vs. "accents-decomposed". Don't know.

DaveMielke commented:
Maybe I'm not supposed to be able to do this, but I'm trying to apply my grc-*
test cases to just the subtable(s). I've removed the locale and assert lines,
and pointed table to the correct file. It seems to work except that the
received lines seem to be containing extra blanks so the test cases fail. How
can I fix that?

-- 
I believe the Bible to be the very Word of God: http://Mielke.cc/bible/
Dave Mielke            | 2213 Fox Crescent | WebHome: http://Mielke.cc/
EMail: Dave@Mielke.cc  | Ottawa, Ontario   | Twitter: @Dave_Mielke
Phone: +1 613 726 0014 | Canada  K2A 1H7   |

bertfrees commented:
Can't say without trying it myself. If you push what you have done I'll fix it.

DaveMielke commented:
[quoted lines by Bert Frees on 2019/07/22 at 15:38 -0700]

Hi:

>Can't say without trying it myself. If you push what you have done I'll fix it.

The attached grc-international.yaml shows what I'm trying to do. For me, it
says 57 tests with 55 failures, so only the ones with no blanks are passing.

-- 
I believe the Bible to be the very Word of God: http://Mielke.cc/bible/
Dave Mielke            | 2213 Fox Crescent | WebHome: http://Mielke.cc/
EMail: Dave@Mielke.cc  | Ottawa, Ontario   | Twitter: @Dave_Mielke
Phone: +1 613 726 0014 | Canada  K2A 1H7   |

# Copyright © 2019 by Dave Mielke <dave@mielke.cc>
#
# Copying and distribution of this file, with or without modification,
# are permitted in any medium without royalty provided the copyright
# notice and this notice are preserved. This file is offered as-is,
# without any warranty.

display: unicode-without-blank.dis
table: grc-international-decomposed.uti
flags: {testmode: forward}
tests:

- # lowercase letters
  - αβγδεζηθικλμνξοπρστυφχψως
  - ⠁⠃⠛⠙⠑⠵⠱⠹⠊⠅⠇⠍⠝⠭⠕⠏⠗⠎⠞⠥⠋⠯⠽⠺⠎

- # uppercase letters
  - ΑΒΓΔΕΖΗΘΙΚΛΜΝΞΟΠΡΣΤΥΦΧΨΩ
  - ⠨⠁⠨⠃⠨⠛⠨⠙⠨⠑⠨⠵⠨⠱⠨⠹⠨⠊⠨⠅⠨⠇⠨⠍⠨⠝⠨⠭⠨⠕⠨⠏⠨⠗⠨⠎⠨⠞⠨⠥⠨⠋⠨⠯⠨⠽⠨⠺

- # accents
  - ᾿ ῾ ΄ ` ´ ῀
  - ⠴ ⠓ ⠈ ⠠ ⠈ ⠐

- # lowercase letters with dasia
  - ἁ ἑ ἡ ἱ ὁ ὑ ὡ ῥ
  - ⠓⠁ ⠓⠑ ⠓⠱ ⠓⠊ ⠓⠕ ⠓⠥ ⠓⠺ ⠗⠓

- # lowercase letters with dasia and oxia
  - ἅ ἕ ἥ ἵ ὅ ὕ ὥ
  - ⠓⠈⠁ ⠓⠈⠑ ⠓⠈⠱ ⠓⠈⠊ ⠓⠈⠕ ⠓⠈⠥ ⠓⠈⠺

- # lowercase letters with dasia and oxia and ypogegrammeni
  - ᾅ ᾕ ᾥ
  - ⠓⠈⠁⠔ ⠓⠈⠱⠔ ⠓⠈⠺⠔

- # lowercase letters with dasia and perispomeni
  - ἇ ἧ ἷ ὗ ὧ
  - ⠓⠐⠁ ⠓⠐⠱ ⠓⠐⠊ ⠓⠐⠥ ⠓⠐⠺

- # lowercase letters with dasia and perispomeni and ypogegrammeni
  - ᾇ ᾗ ᾧ
  - ⠓⠐⠁⠔ ⠓⠐⠱⠔ ⠓⠐⠺⠔

- # lowercase letters with dasia and varia
  - ἃ ἓ ἣ ἳ ὃ ὓ ὣ
  - ⠓⠠⠁ ⠓⠠⠑ ⠓⠠⠱ ⠓⠠⠊ ⠓⠠⠕ ⠓⠠⠥ ⠓⠠⠺

- # lowercase letters with dasia and varia and ypogegrammeni
  - ᾃ ᾓ ᾣ
  - ⠓⠠⠁⠔ ⠓⠠⠱⠔ ⠓⠠⠺⠔

- # lowercase letters with dasia and ypogegrammeni
  - ᾁ ᾑ ᾡ
  - ⠓⠁⠔ ⠓⠱⠔ ⠓⠺⠔

- # lowercase letters with dialytika
  - ϊ ϋ
  - ⠊ ⠥

- # lowercase letters with dialytika and oxia
  - ΐ ΰ
  - ⠈⠊ ⠈⠥

- # lowercase letters with dialytika and perispomeni
  - ῗ ῧ
  - ⠐⠊ ⠐⠥

- # lowercase letters with dialytika and tonos
  - ΐ ΰ
  - ⠈⠊ ⠈⠥

- # lowercase letters with dialytika and varia
  - ῒ ῢ
  - ⠠⠊ ⠠⠥

- # lowercase letters with macron
  - ᾱ ῑ ῡ
  - ⠁ ⠊ ⠥

- # lowercase letters with oxia
  - ά έ ή ί ό ύ ώ
  - ⠈⠁ ⠈⠑ ⠈⠱ ⠈⠊ ⠈⠕ ⠈⠥ ⠈⠺

- # lowercase letters with oxia and ypogegrammeni
  - ᾴ ῄ ῴ
  - ⠈⠁⠔ ⠈⠱⠔ ⠈⠺⠔

- # lowercase letters with perispomeni
  - ᾶ ῆ ῖ ῦ ῶ
  - ⠐⠁ ⠐⠱ ⠐⠊ ⠐⠥ ⠐⠺

- # lowercase letters with perispomeni and ypogegrammeni
  - ᾷ ῇ ῷ
  - ⠐⠁⠔ ⠐⠱⠔ ⠐⠺⠔

- # lowercase letters with psili
  - ἀ ἐ ἠ ἰ ὀ ὐ ὠ ῤ
  - ⠴⠁ ⠴⠑ ⠴⠱ ⠴⠊ ⠴⠕ ⠴⠥ ⠴⠺ ⠴⠗

- # lowercase letters with psili and oxia
  - ἄ ἔ ἤ ἴ ὄ ὔ ὤ
  - ⠴⠈⠁ ⠴⠈⠑ ⠴⠈⠱ ⠴⠈⠊ ⠴⠈⠕ ⠴⠈⠥ ⠴⠈⠺

- # lowercase letters with psili and oxia and ypogegrammeni
  - ᾄ ᾔ ᾤ
  - ⠴⠈⠁⠔ ⠴⠈⠱⠔ ⠴⠈⠺⠔

- # lowercase letters with psili and perispomeni
  - ἆ ἦ ἶ ὖ ὦ
  - ⠴⠐⠁ ⠴⠐⠱ ⠴⠐⠊ ⠴⠐⠥ ⠴⠐⠺

- # lowercase letters with psili and perispomeni and ypogegrammeni
  - ᾆ ᾖ ᾦ
  - ⠴⠐⠁⠔ ⠴⠐⠱⠔ ⠴⠐⠺⠔

- # lowercase letters with psili and varia
  - ἂ ἒ ἢ ἲ ὂ ὒ ὢ
  - ⠴⠠⠁ ⠴⠠⠑ ⠴⠠⠱ ⠴⠠⠊ ⠴⠠⠕ ⠴⠠⠥ ⠴⠠⠺

- # lowercase letters with psili and varia and ypogegrammeni
  - ᾂ ᾒ ᾢ
  - ⠴⠠⠁⠔ ⠴⠠⠱⠔ ⠴⠠⠺⠔

- # lowercase letters with psili and ypogegrammeni
  - ᾀ ᾐ ᾠ
  - ⠴⠁⠔ ⠴⠱⠔ ⠴⠺⠔

- # lowercase letters with tonos
  - ά έ ή ί ό ύ ώ
  - ⠈⠁ ⠈⠑ ⠈⠱ ⠈⠊ ⠈⠕ ⠈⠥ ⠈⠺

- # lowercase letters with varia
  - ὰ ὲ ὴ ὶ ὸ ὺ ὼ
  - ⠠⠁ ⠠⠑ ⠠⠱ ⠠⠊ ⠠⠕ ⠠⠥ ⠠⠺

- # lowercase letters with varia and ypogegrammeni
  - ᾲ ῂ ῲ
  - ⠠⠁⠔ ⠠⠱⠔ ⠠⠺⠔

- # lowercase letters with vrachy
  - ᾰ ῐ ῠ
  - ⠁ ⠊ ⠥

- # lowercase letters with ypogegrammeni
  - ᾳ ῃ ῳ
  - ⠁⠔ ⠱⠔ ⠺⠔

- # uppercase letters with dasia
  - Ἁ Ἑ Ἡ Ἱ Ὁ Ὑ Ὡ Ῥ
  - ⠨⠓⠁ ⠨⠓⠑ ⠨⠓⠱ ⠨⠓⠊ ⠨⠓⠕ ⠨⠓⠥ ⠨⠓⠺ ⠨⠗⠓

- # uppercase letters with dasia and oxia
  - Ἅ Ἕ Ἥ Ἵ Ὅ Ὕ Ὥ
  - ⠨⠓⠈⠁ ⠨⠓⠈⠑ ⠨⠓⠈⠱ ⠨⠓⠈⠊ ⠨⠓⠈⠕ ⠨⠓⠈⠥ ⠨⠓⠈⠺

- # uppercase letters with dasia and oxia and prosgegrammeni
  - ᾍ ᾝ ᾭ
  - ⠨⠓⠈⠁⠔ ⠨⠓⠈⠱⠔ ⠨⠓⠈⠺⠔

- # uppercase letters with dasia and perispomeni
  - Ἇ Ἧ Ἷ Ὗ Ὧ
  - ⠨⠓⠐⠁ ⠨⠓⠐⠱ ⠨⠓⠐⠊ ⠨⠓⠐⠥ ⠨⠓⠐⠺

- # uppercase letters with dasia and perispomeni and prosgegrammeni
  - ᾏ ᾟ ᾯ
  - ⠨⠓⠐⠁⠔ ⠨⠓⠐⠱⠔ ⠨⠓⠐⠺⠔

- # uppercase letters with dasia and prosgegrammeni
  - ᾉ ᾙ ᾩ
  - ⠨⠓⠁⠔ ⠨⠓⠱⠔ ⠨⠓⠺⠔

- # uppercase letters with dasia and varia
  - Ἃ Ἓ Ἣ Ἳ Ὃ Ὓ Ὣ
  - ⠨⠓⠠⠁ ⠨⠓⠠⠑ ⠨⠓⠠⠱ ⠨⠓⠠⠊ ⠨⠓⠠⠕ ⠨⠓⠠⠥ ⠨⠓⠠⠺

- # uppercase letters with dasia and varia and prosgegrammeni
  - ᾋ ᾛ ᾫ
  - ⠨⠓⠠⠁⠔ ⠨⠓⠠⠱⠔ ⠨⠓⠠⠺⠔

- # uppercase letters with dialytika
  - Ϊ Ϋ
  - ⠨⠊ ⠨⠥

- # uppercase letters with macron
  - Ᾱ Ῑ Ῡ
  - ⠨⠁ ⠨⠊ ⠨⠥

- # uppercase letters with oxia
  - Ά Έ Ή Ί Ό Ύ Ώ
  - ⠨⠈⠁ ⠨⠈⠑ ⠨⠈⠱ ⠨⠈⠊ ⠨⠈⠕ ⠨⠈⠥ ⠨⠈⠺

- # uppercase letters with prosgegrammeni
  - ᾼ ῌ ῼ
  - ⠨⠁⠔ ⠨⠱⠔ ⠨⠺⠔

- # uppercase letters with psili
  - Ἀ Ἐ Ἠ Ἰ Ὀ Ὠ
  - ⠨⠴⠁ ⠨⠴⠑ ⠨⠴⠱ ⠨⠴⠊ ⠨⠴⠕ ⠨⠴⠺

- # uppercase letters with psili and oxia
  - Ἄ Ἔ Ἤ Ἴ Ὄ Ὤ
  - ⠨⠴⠈⠁ ⠨⠴⠈⠑ ⠨⠴⠈⠱ ⠨⠴⠈⠊ ⠨⠴⠈⠕ ⠨⠴⠈⠺

- # uppercase letters with psili and oxia and prosgegrammeni
  - ᾌ ᾜ ᾬ
  - ⠨⠴⠈⠁⠔ ⠨⠴⠈⠱⠔ ⠨⠴⠈⠺⠔

- # uppercase letters with psili and perispomeni
  - Ἆ Ἦ Ἶ Ὦ
  - ⠨⠴⠐⠁ ⠨⠴⠐⠱ ⠨⠴⠐⠊ ⠨⠴⠐⠺

- # uppercase letters with psili and perispomeni and prosgegrammeni
  - ᾎ ᾞ ᾮ
  - ⠨⠴⠐⠁⠔ ⠨⠴⠐⠱⠔ ⠨⠴⠐⠺⠔

- # uppercase letters with psili and prosgegrammeni
  - ᾈ ᾘ ᾨ
  - ⠨⠴⠁⠔ ⠨⠴⠱⠔ ⠨⠴⠺⠔

- # uppercase letters with psili and varia
  - Ἂ Ἒ Ἢ Ἲ Ὂ Ὢ
  - ⠨⠴⠠⠁ ⠨⠴⠠⠑ ⠨⠴⠠⠱ ⠨⠴⠠⠊ ⠨⠴⠠⠕ ⠨⠴⠠⠺

- # uppercase letters with psili and varia and prosgegrammeni
  - ᾊ ᾚ ᾪ
  - ⠨⠴⠠⠁⠔ ⠨⠴⠠⠱⠔ ⠨⠴⠠⠺⠔

- # uppercase letters with tonos
  - Ά Έ Ή Ί Ό Ύ Ώ
  - ⠨⠈⠁ ⠨⠈⠑ ⠨⠈⠱ ⠨⠈⠊ ⠨⠈⠕ ⠨⠈⠥ ⠨⠈⠺

- # uppercase letters with varia
  - Ὰ Ὲ Ὴ Ὶ Ὸ Ὺ Ὼ
  - ⠨⠠⠁ ⠨⠠⠑ ⠨⠠⠱ ⠨⠠⠊ ⠨⠠⠕ ⠨⠠⠥ ⠨⠠⠺

- # uppercase letters with vrachy
  - Ᾰ Ῐ Ῠ
  - ⠨⠁ ⠨⠊ ⠨⠥

bertfrees commented:
The following will fix the test:

```yaml
table: {locale: grc}
table: spaces.uti,grc-international-decomposed.uti
```

I've added "spaces.uti", and I also kept the original table (the "utb" one) in the test because in the end that is what we are interested in so it should be tested as well. All the tests in the table are tested against all the tables that you defined at the beginning of the file.

DaveMielke commented:
[quoted lines by Bert Frees on 2019/07/23 at 03:17 -0700]

>The following will fix the test:

Thanks. Now they do indeed work. This means that I can now start working on the
composed version.

>I've added "spaces.uti", and I also kept the original table (the "utb" one) in
>the test because in the end that is what we are interested in so it should be
>tested as well. All the tests in the table are tested against all the tables
>that you defined at the beginning of the file.

What's picking up the .utb? The {locale: grc}? I'm only getting one SUCCESS
line - not two. Is that how it works?

-- 
I believe the Bible to be the very Word of God: http://Mielke.cc/bible/
Dave Mielke            | 2213 Fox Crescent | WebHome: http://Mielke.cc/
EMail: Dave@Mielke.cc  | Ottawa, Ontario   | Twitter: @Dave_Mielke
Phone: +1 613 726 0014 | Canada  K2A 1H7   |

bertfrees commented:
> What's picking up the .utb? The {locale: grc}? I'm only getting one SUCCESS line - not two. Is that how it works?

Yes. That's just how it works.

bertfrees commented:
> what, from your perspective, still needs to be done (for both PRs)

Can't think of anything right now. Looks complete to me. I'm going to have a final look at it and add a few commits myself, like updating the NEWS file, the Makefiles and the display-names file, and maybe do some additions to the metadata and comments in the headers. Then I'll let you review my commits if you want and maybe let Christian have a look and then it's ready for the 3.11 release.
